GeneralsMD/Code/GameEngine/Source/GameNetwork/GameSpy/Thread/BuddyThread.cpp

```diff
@@ -557,8 +558,11 @@ void BuddyThreadClass::connectCallback( GPConnection *con, GPConnectResponseArg
 				DEBUG_LOG(("User Error: Create Account instead of Login.  Fixing them..."));
 				BuddyRequest req;
 				req.buddyRequestType = BuddyRequest::BUDDYREQUEST_LOGIN;
+				static_assert(ARRAY_SIZE(req.arg.login.nick) > 0, "Array size too small");
 				strcpy(req.arg.login.nick, m_nick.c_str());
+				static_assert(ARRAY_SIZE(req.arg.login.email) > 0, "Array size too small");
 				strcpy(req.arg.login.email, m_email.c_str());
+				static_assert(ARRAY_SIZE(req.arg.login.password) > 0, "Array size too small");
 				strcpy(req.arg.login.password, m_pass.c_str());
```

Comment:
strlcpy instead of static_assert, because size of string is indeterminate. 3 times